app/src/main/java/org/mozilla/tv/firefox/navigationoverlay/BrowserNavigationOverlay.kt

```diff
@@ -238,24 +240,26 @@ class BrowserNavigationOverlay @JvmOverloads constructor(
         }
 
         pocketVideoMegaTileView.setOnClickListener(this)
+    }
 
-        if (pocketVideos.isCompleted) {
-            pocketVideoMegaTileView.pocketVideos = pocketVideos.getCompleted()
-
-            if (pocketVideoMegaTileView.pocketVideos == null) {
-                showMegaTileError()
-            } else {
-                pocketVideosContainer.visibility = View.VISIBLE
-                pocketErrorContainer.visibility = View.GONE
-                pocketVideosContainer.contentDescription = resources.getString(R.string.pocket_home_a11y_tile_focused,
-                        resources.getString(R.string.pocket_brand_name))
-            }
-        } else {
-            // TODO: #864 show loading screen
-            launch(UI) {
-                pocketVideoMegaTileView.pocketVideos = pocketVideos.await()
+    fun observeForMegaTile(fragment: Fragment) {
+        // TODO remove this
+        // This function is necessary because BrowserNavigationOverlay is not a LifecycleOwner,
+        // and so cannot subscribe to PocketRepo's LiveData. When the overlay is turned into a
+        // fragment, this can be moved into the overlay fragment
+        val serviceLocator = fragment.context!!.serviceLocator
+        val factory = ViewModelFactory(serviceLocator)
```

Comment:
`ViewModelFactory` is stateless: should we just store one instance in the serviceLocator instead to make this (neglibly) more efficient?